test/solution-template-validation-tests/test/mainTemplateTests.js

```diff
@@ -115,16 +117,17 @@ describe('mainTemplate.json file - ', () => {
             });
         });
 
-        it('apiVersions must NOT be retrieved using providers().apiVersions[n].  This function is non-deterministic', () => {
+        // providers().apiVersions[n] MUST not be present
+        describe('apiVersions must NOT be retrieved using providers().apiVersions[n].  This function is non-deterministic', () => {
             templateFileJSONObjects.forEach(templateJSONObject => {
                 var templateObject = templateJSONObject.value;
-                templateObject.should.have.property('resources');
                 var message = 'in file:' + templateJSONObject.filename + ' should NOT have api version determined by providers().';
-                Object.keys(templateObject.resources).forEach(resource => {
-                    var val = templateObject.resources[resource];
-                    if (val.apiVersion) {
-                        val.apiVersion.should.withMessage(getErrorMessage(val, templateJSONObject.filepath, message)).not.contain('providers(');
-                    }
+                var properties = Object.keys(templateObject);
+                var resources = templateObject.resources;
+                it.each(properties, "providers().apiVersions[n] must NOT be present in the template", function(element, next) {
```

Comment:
can you call out the element in the description, so we know what we're testing against? Look at [this ](https://github.com/Azure/azure-quickstart-templates/blob/f94add6ac253e25f8d852095c500fcffc493432f/test/solution-template-validation-tests/test/createUiDefTests.js#L92)for example.

Reply:
This is a little hard, because the properties are all properties in the json, so the key for each element is hard to tell.

Reply:
How does the output look like, if we don't call out the element? Can you paste it here?

Reply:
As discussed, please change this back to "it", but catch all occurrences of the offending string. Also, the error message should indicate the line number